test/Wallet.test.ts

```diff
@@ -54,20 +59,210 @@ describe("Wallet", () => {
+      expect(() => wallet.getRedeemScript("bitcoincash:qrsy0xwugcajsqa99c9nf05pz7ndckj55ctlsztu2p"))
+      .toThrow(IllegalArgumentException)
+    })
+    it("should throws ProviderException if the wallet provider throws an error.", async () => {
+      walletProvider = new (jest.fn<IWalletProvider>(() => ({
+        getRedeemScripts: jest.fn(() => Promise.reject())
+      })))()
+      wallet = new Wallet(new Providers(undefined, walletProvider))
+      await expect(wallet.getRedeemScript("bitcoincash:pzwpv4lm29pv4pdt95n74prlvj8vzu4qzg7pgrspya"))
+      .rejects.toThrow(ProviderException)
+    })
+    it("should throws ProviderException if the wallet provider invalid value.", async () => {
+      walletProvider = new (jest.fn<IWalletProvider>(() => ({
+        getRedeemScripts: jest.fn(() => Promise.resolve([]))
+      })))()
+      wallet = new Wallet(new Providers(undefined, walletProvider))
+      await expect(wallet.getRedeemScript("bitcoincash:pzwpv4lm29pv4pdt95n74prlvj8vzu4qzg7pgrspya"))
+      .rejects.toThrow(ProviderException)
+    })
+  })
 
   //
   // getRedeemScripts
   //
+  describe("getRedeemScripts()", () => {
+    beforeEach(() => {
+      const networkProvider = new (jest.fn<INetworkProvider>())()
+      walletProvider = new (jest.fn<IWalletProvider>(() => ({
+        getRedeemScripts: jest.fn(() => Promise.resolve(["03424f587e06424954424f5887", "789787a72c21452a1c98ff"]))
+      })))()
+      const providers = new Providers(networkProvider, walletProvider)
+      wallet = new Wallet(providers)
+    })
+
+    it("should be success if there is no problem.", async () => {
+      await wallet.getRedeemScripts()
+    })
+    it("should calls IWalletProvider#getRedeemScripts", async () => {
+      await wallet.getRedeemScripts()
+      expect(walletProvider.getRedeemScripts).toBeCalled()
+    })
+    it("should return the same value as IWalletProvider#getRedeemScripts", async () => {
+      const sciprts = await wallet.getRedeemScripts()
+      expect(sciprts).toBe(["03424f587e06424954424f5887", "789787a72c21452a1c98ff"])
+    })
+    it("should throws ProviderException if the wallet provider throws an error.", async () => {
+      walletProvider = new (jest.fn<IWalletProvider>(() => ({
+        getRedeemScripts: jest.fn(() => Promise.reject())
+      })))()
+      wallet = new Wallet(new Providers(undefined, walletProvider))
+      await expect(wallet.getRedeemScripts()).rejects.toThrow(ProviderException)
+    })
+    it("should throws ProviderException if the wallet provider invalid value.", async () => {
+      walletProvider = new (jest.fn<IWalletProvider>(() => ({
+        getRedeemScripts: jest.fn(() => Promise.resolve(1))
+      })))()
+      wallet = new Wallet(new Providers(undefined, walletProvider))
+      await expect(wallet.getRedeemScripts()).rejects.toThrow(ProviderException)
+    })
+  })
 
   //
   // addRedeemScript
   //
+  describe("addRedeemScript()", () => {
```

Comment:
I think the test that "should throws IllegalArgumentException if the redeenScritpt is empty" needed.